qmail-smtpd.c

```diff
@@ -230,7 +233,12 @@ void smtp_helo(arg) char *arg;
 }
 void smtp_ehlo(arg) char *arg;
 {
-  smtp_greet("250-"); out("\r\n250-PIPELINING\r\n250 8BITMIME\r\n");
+  unsigned int i;
+  smtp_greet("250-");
+  for (i = 0; ehlo_gen[i]; i++) {
+    ehlo_gen[i]();
+  }
+  out("\r\n250-PIPELINING\r\n250 8BITMIME\r\n");
```

Comment:
If we print `\r\n` at the end of each line, we can remove this leading `\r\n` here, and we are on our foot.